File: aricode/sparring/challenges/08_gcd/gcd.c
/* C - Challenge 08: Euclidean GCD */
/* Compute GCD(462, 1071) = 21 */

int gcd(int a, int b) {
    while (b > 0) {
        int t = b;
        b = a % b;
        a = t;
    }
    return a;
}

int main(void) {
    return gcd(462, 1071);
}

File: aricode/sparring/challenges/08_gcd/gcd.rs
// Rust - Challenge 08: Euclidean GCD
// Compute GCD(462, 1071) = 21
use std::process::ExitCode;

fn gcd(mut a: i32, mut b: i32) -> i32 {
    while b > 0 {
        let t = b;
        b = a % b;
        a = t;
    }
    a
}

fn main() -> ExitCode {
    ExitCode::from(gcd(462, 1071) as u8)
}
